app/full_text_search.py

```diff
@@ -0,0 +1,64 @@
+from const import PERSON_FULLTEXT_INDEX_NAME
+
+from google.appengine.api import search
+import re
+import model
+import logging
+
+def search_with_index(repo, query_obj, max_results):
+
+    def create_query(query):
+        return re.sub(r' ', ' AND ', query)
```

Comment:
Does this work for a query which contains double whitespace (e.g., "hoge  foo") or a query which starts or ends with a whitespace (e.g., " hoge foo ")?